src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/PluginImpl.java

```diff
@@ -420,12 +431,22 @@ public static void save_() throws IOException {
         plugin.save();
     }
 
+    // Reading from the location where the data used to be back when this implemented hudson.Plugin
+    @Override
+    protected XmlFile getConfigFile() {
+        return new XmlFile(Jenkins.XSTREAM, new File(Jenkins.get().getRootDir(), "gerrit-trigger.xml"));
```

Comment:
I had a look around and the only plugin I could find searching https://github.com/search?l=&p=7&q=org%3Ajenkinsci+gerrit.trigger.PluginImpl+fork%3Atrue&ref=advsearch&type=Code

was build failure analyzer and that seems to be using a specific extension point https://github.com/jenkinsci/build-failure-analyzer-plugin/blob/bf0ab28e79ff74a4e9447ba64d031f041c9cc30f/src/main/java/com/sonyericsson/jenkins/plugins/bfa/GerritMessageProviderExtension.java#L44

so this seems safe to me but I could be missing something

Reply:
Good point in searching the possible impact on other plugins, it would be worth retesting whether we do not break https://github.com/jenkinsci/configuration-as-code-groovy-plugin/blob/78d325d813c33b8515045b7267355b8624cd4508/demos/gerrit-trigger.yml.

Following appears safe too: https://github.com/jenkinsci/sonar-gerrit-plugin/blob/master/src/main/java/org/jenkinsci/plugins/sonargerrit/config/GerritAuthenticationConfig.java

Reply:
The first will definitely break but groovy compatibility isn’t normally as much of an issue

Reply:
The first did break, eg. the Jenkins.instance.getPlugin(...) call returns `null`. Is there an another way to do this now, or should we pin 2.30.5 until https://issues.jenkins-ci.org/browse/JENKINS-53250 has been implemented?

Reply:
@Gustra, `PluginImpl.getInstance()` should work before and after this fix.

Reply:
Thank you, @olivergondza. Works like a charm.